app/uk/gov/hmrc/agentsubscription/controllers/RegistrationController.scala

```diff
@@ -20,36 +20,27 @@ import javax.inject._
 
 import play.api.libs.json.Json
 import play.api.libs.json.Json.toJson
-import play.api.mvc._
-import uk.gov.hmrc.agentsubscription._
 import uk.gov.hmrc.agentsubscription.auth.AuthActions
-import uk.gov.hmrc.agentsubscription.connectors.{AuthConnector, DesConnector, DesIndividual, DesRegistrationResponse}
-import uk.gov.hmrc.agentsubscription.model.{RegistrationDetails, Utr, postcodeWithoutSpacesRegex}
-import uk.gov.hmrc.play.http.HeaderCarrier
+import uk.gov.hmrc.agentsubscription.connectors.AuthConnector
+import uk.gov.hmrc.agentsubscription.model.{Utr, postcodeWithoutSpacesRegex}
+import uk.gov.hmrc.agentsubscription.service.RegistrationService
 import uk.gov.hmrc.play.http.logging.MdcLoggingExecutionContext._
 import uk.gov.hmrc.play.microservice.controller.BaseController
 
 import scala.concurrent.Future
 
 @Singleton
-class RegistrationController @Inject()(val desConnector: DesConnector, override val authConnector: AuthConnector )
+class RegistrationController @Inject()(service: RegistrationService, override val authConnector: AuthConnector)
   extends BaseController with AuthActions {
 
   def getRegistration(utr: String, postcode: String) = withAgentAffinityGroup.async { implicit request =>
-      if (Utr.isValid(utr) && validPostcode(postcode))
-        getRegistrationFromDes(utr, postcode)
-      else
-        Future successful BadRequest(Json.obj("code" -> "INVALID_UTR"))
-  }
-
-  private def getRegistrationFromDes(utr: String, postcode: String)(implicit hc: HeaderCarrier): Future[Result] = {
-    desConnector.getRegistration(utr) map {
-      case Some(DesRegistrationResponse(Some(desPostcode), isAnASAgent, organisationName, None)) if postcodesMatch(desPostcode, postcode) =>
-        Ok(toJson(RegistrationDetails(isAnASAgent, organisationName)))
-      case Some(DesRegistrationResponse(Some(desPostcode), isAnASAgent, _, Some(DesIndividual(first, last)))) if postcodesMatch(desPostcode, postcode) =>
-        Ok(toJson(RegistrationDetails(isAnASAgent, Some(s"$first $last"))))
-      case _ => NotFound
-    }
+    //TODO code will be INVALID_UTR if UTR is *valid* but postcode is invalid - this doesn't make sense
```

Comment:
I added this TODO because I noticed what looks like a pre-existing bug.
I'll look into that separately.